code/game/objects/items/toys.dm

```diff
@@ -1032,45 +1032,73 @@
 	var/cooldown = 0
 
 /obj/item/toy/nuke/attack_self(mob/user)
-	if (cooldown < world.time)
-		cooldown = world.time + 1800 //3 minutes
+	var/timeleft = (cooldown - world.time)
+	if (obj_flags & EMAGGED && cooldown < world.time)
+		cooldown = world.time + 600
+		user.visible_message("<span class='hear'>You hear the click of a button.</span>", "<span class='notice'>You activate [src], it plays a loud noise!</span>")
+		sleep(5)
```

Comment:
you have to check for item qdeleted afterevery time you sleep, how can people not know this by now

Reply:
Because I left for like a year and this is the first bit of code I've used with a sleep in it. What do I need to add to un-fuck it?